nixos/modules/services/networking/6tunnel.nix

```diff
@@ -0,0 +1,84 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services._6tunnel;
+  user = cfg.user;
+
+in {
+  options.services._6tunnel = {
+
+    user = mkOption {
+      default = "6tunnel";
+      description = "User that runs 6tunnel";
+      type = types.str;
+    };
+
+    tunnels = mkOption {
+      default = [];
+      description = "Which tunnels to create.";
+      type = with types; listOf (submodule {
+        options = {
+          port = mkOption {
+            description = "Local port to open.";
+            type = types.port;
+          };
+          host = mkOption {
+            description = "Remote host to tunnel to.";
+            type = types.str;
+          };
+          remotePort = mkOption {
+            default = null;
+            description = "Port on remote host. If null (default), it uses the the local port.";
+            type = with types; nullOr port;
+          };
+          restart = mkOption {
+            default = null;
+            description = ''
+              Whether to restart the tunnel periodically, which is useful for non-static IP addresses.
+              The format is described in SYSTEMD.TIME(7).
+
+              If null (default), the tunnel won't restart periodically.
+            '';
+            type = with types; nullOr str;
+          };
+        };
+      });
+      example = [
+        { port = "8080"; host = "192.168.0.2"; remotePort = 80; }
+        { port = "443"; host = "www.example.com"; }
+        { port = "8053"; host = "dynamicip.example.com"; restart = "1h"; }
+      ];
+    };
+
+  };
+
+  config = mkIf (cfg.tunnels != {}) {
+
+    networking.firewall.allowedTCPPorts = map (tunnel: tunnel.port) cfg.tunnels;
+
+    systemd.services = listToAttrs (map (tunnel:
+      nameValuePair "6tunnel-${toString tunnel.port}" {
+        description = "6tunnel from ${toString tunnel.port} to ${tunnel.host}";
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Restart = "always";
+          RuntimeMaxSec = mkIf (tunnel.restart != null) tunnel.restart;
+          User = user;
```

Comment:
`Group` should be set as well.

Is there any reason to make the user configurable and not use `DynamicUser`?

Reply:
> `Group` should be set as well.
Can I just set it to `nogroup`?

> Is there any reason to make the user configurable and not use `DynamicUser`?
IIRC, if you want to port to ports lower than some number (I'm unsure which one), you'd need root privileges, don't you?
`DynamicUser` is actually a great concept, I didn't know it existed. Thanks!

Reply:
You can use `AmbientCapabilities = "CAP_NET_BIND_SERVICE";` with `DynamicUser = true;` to allow binding to any port.

Reply:
Great, thank you.